Adding additional Sokal 1998 calculations for Moran I_i (reflow)

jeffcsauer opened the pull request:
Apologies for the double PR. A spacing issue on my end made it seem the entire file had been changed. 

_Same PR information from https://github.com/r-spatial/spdep/pull/57_

This PR adds additional Sokal 1998 calculations to the spdep localmoran function. These calculations distinguish between 'conditional' and 'total' randomization hypotheses (e.g., whether or not a value of z at site i is fixed).

The PR adds a TRUE/FALSE argument called conditional. Defaulted to conditional=TRUE, E.Ii and Var.Ii results are calculated using Sokal 1998 Eqs 7 and 8. If conditional=FALSE, E.Ii and Var.Ii columns are calculated using the existing Sokal 1998 Eqs 4 and 5. Ii, Z.Ii, and Pr(z > 0) are calculated as usual. This preserves the existing results structure of localmoran and avoids adding additional columns.

The localmoran.Rd file has also been updated. These changes have not yet been extended to the new localmoran_perm function.

This PR was built with @ljwolf.

Open to suggestions and guidance on the preferred way to incorporate these changes!

ljwolf commented on R/localmoran.R:
```diff
@@ -1,7 +1,7 @@
 #
 
-localmoran <- function(x, listw, zero.policy=NULL, na.action=na.fail, 
+localmoran <- function(x, listw, zero.policy=NULL, na.action=na.fail, conditional=TRUE, 
```
This'll probably have to be false for the time being, as that'd change the default output...

rsbivand replied:
Agreed, the usual approach is not to change something that isn't a bug as such, but a consequence of an implementation choice, which can now be made by the user. We can revisit this later on.

jeffcsauer replied:
Changed default to FALSE.

rsbivand reviewed (commented):
Great! Could you please add a reference to the article draft to the man page? If you both agree, I'd like to add you to DESCRIPTION as contributors, is that OK?

jeffcsauer commented:
Updated `localmoran` documentation to to reflect change in default `conditional` value and added citation for the preprint. 

Perfectly fine with me!
